### ISSUE TITLE
Build a basic predictive GNN model using PyG to predict overall survival(months) for patients

### ISSUE THREAD

**Issue by Favourj-bit** [cannin/gsoc_2023_pytorch_pathway_commons#10]
https://colab.research.google.com/drive/1YHTJgRreJnAmvs3qNkErBFc41vAwqSa9?usp=sharing

**Comment by Favourj-bit** [comment]
![image](https://github.com/cannin/gsoc_2023_pytorch_pathway_commons/assets/63251266/68b55cfe-964a-4d25-992c-fc75c742aeee)
@inoue0426 , pls i keep getting this error and i cannot seem to figure it out.
this is the code before it: 
![image](https://github.com/cannin/gsoc_2023_pytorch_pathway_commons/assets/63251266/c79eba34-d9fa-485b-aedf-92855c61526b)
I'm trying to create the data structure using the data from cBio and pathway common. I'm wondering if you could help me give it a look. I've attached the link to the colab notebook above

**Comment by inoue0426** [comment]
@Favourj-bit 

You may need to follow this model.
https://colab.research.google.com/drive/1I8a0DfQ3fI7Njc62__mVXUlcAleUclnb?usp=sharing

The current structure is not suitable for this problem maybe.
This is the sample code.
https://colab.research.google.com/drive/1MG09YsUhnINSnFVx4uv_nlgJHuS4ZKwE?usp=sharing

You may need to make the graph which for each patient.
I mean 78 Graphes and use it for Graph Regression like Graph Classification.

**Comment by Favourj-bit** [comment]
ok, thanks. Do you think you could give my notebook a look and let me know what you think of the steps i took up to the splitting of the dataset? That is where i keep getting errors no matter what I do.

**Comment by inoue0426** [comment]
@Favourj-bit So maybe the indexing of the data loader is incorrect. I'm not sure what is 2 here but maybe it's an index of data and it doesn't match to the data. Anyways, you don't need to use the torch's data loader, you may need to use Graph Data Loader sometime.

**Comment by Favourj-bit** [comment]
@inoue0426 
I have been able to create patients specific graphs. there is also a feature in PyG that allows me to match them to one batch. However, I'm getting some errors in this aspect. I'm still debugging it. I'll share it here if i can figure it out before our meeting today

**Comment by Favourj-bit** [comment]
@inoue0426 @cannin 
I've been able to implement a very basic model. The loss is a little big, I'm using MSE. I hope you could give it a look, then we could discuss on the next steps.
https://colab.research.google.com/drive/1YHTJgRreJnAmvs3qNkErBFc41vAwqSa9?usp=sharing

**Comment by inoue0426** [comment]
@Favourj-bit Could you add this library and upload jupyter notebook instead of colab?
https://www.reviewnb.com/

**Comment by inoue0426** [comment]
@Favourj-bit Below is my comment. Overall looks good. Loss is decreasing, just the epoch is too small.
1. You don't need to use a small batch like 16. Could you try to use just ```data_train```, and ```data_test``` as an input?
2. Epoch is too small, at least 100.
3. Should add dropout, GraphNorm. Also dropout rate is too large, firstly try to use 0.2
4. I'm not sure that global_meaning_pool is working here. Just use x as an input of linear might be better.
5. I've never seen model.double(). I think you don't need to do this.
6. The results which show the predictions are not meaningful.
7. Here, you don't have the validation data, so you can just remove the last two cells anyways.

**Comment by Favourj-bit** [comment]
@inoue0426 
1. Okay, running it without batches already. I used 10 epochs to just test the model at first, I knew it was too small but I wanted to see if the model works. I'm running with 100 epochs now. But it has been running for up to 3 hours now and it is not done
2. I don't understand what you mean by this: Should add dropout, GraphNorm
3. The model.double() was done to chane the model parameters to double. ithout doing it, i keep on geeting the type errors with the code. Even when i changed the type of the dataset. It was suggested to me by chatgpt when i was debbugging the error and it worked, that is why it is there.

**Comment by inoue0426** [comment]
@Favourj-bit Looks like you don't use GPU, so it's slow. need to set the device.
https://pytorch.org/docs/stable/generated/torch.cuda.device.html
https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.norm.GraphNorm.html
https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html

**Comment by inoue0426** [comment]
@Favourj-bit I think you already can run the model, so you should try a more extensive dataset and do train, val, and test split like 6:2:2 and then show the test result to ensure the model is working well.
